core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -300,10 +300,19 @@ object LogConfig {
   }
 
   /**
+    * Check that the property values are valid relative to each other
+    */
+  def validateValues(props: Properties) {
+    require(props.getProperty(SegmentBytesProp).toLong <= props.getProperty(RetentionBytesProp).toLong,
+      "segment.bytes "+props.getProperty(SegmentBytesProp)+" is not less than or equal to retention.bytes "+props.getProperty(RetentionBytesProp))
```

Comment:
nitpick: String interpolation reads a little nicer. It might also help to provide a short sentence explaining the restriction.